substrate-node/pallets/pallet-dao/src/lib.rs

```diff
@@ -211,6 +211,17 @@ pub mod pallet {
                 !<ProposalOf<T>>::contains_key(proposal_hash),
                 Error::<T>::DuplicateProposal
             );
+            
+            let now = frame_system::Pallet::<T>::block_number();
```

Comment:
We should check all input before changing the state of the object. assert_noop was failing because the state of the object was changed before throwing an error.